src/compiler/utilities.ts

```diff
@@ -2083,6 +2083,12 @@ namespace ts {
         return start <= textSpanEnd(span) && end >= span.start;
     }
 
+    export function textSpanIntersectsWith2(start1: number, length1: number, start2: number, length2: number) {
+        let end1 = start1 + length1;
+        let end2 = start2 + length2;
+        return start2 <= end1 && end2 >= start1;

```

Comment:
I'm not sure I see the gain in nominal over relational consistency. Maybe just leave a comment like

```TypeScript
// Covers
// (1)
//        s1-----------e1
//              s2-----------e2
//
// (2)
//              s1-----------e1
//        s2-----------e2
//
// (3)
//        s1-----------e1
//            s2---e2
//
// (4)
//            s1---e1
//        s2-----------e2
```

or if you don't see huge utility in it, don't.

Also this might be better called "overlaps with" instead of "intersects with"